js/src/components/paid-ads/asset-group/asset-field.js

```diff
@@ -0,0 +1,130 @@
+/**
+ * External dependencies
+ */
+import classnames from 'classnames';
+import { __, _n, sprintf } from '@wordpress/i18n';
+import { useReducedMotion } from '@wordpress/compose';
+import {
+	useState,
+	useRef,
+	useImperativeHandle,
+	forwardRef,
+} from '@wordpress/element';
+import { chevronUp, chevronDown } from '@wordpress/icons';
+import { Pill } from '@woocommerce/components';
+
+/**
+ * Internal dependencies
+ */
+import AppButton from '.~/components/app-button';
+import HelpPopover from '.~/components/help-popover';
+import './asset-field.scss';
+
+/**
+ * @typedef {Object} AssetFieldHandler
+ * @property {() => void} scrollIntoComponent Scroll to the nearest alignable position to show this component in the visible area.
+ */
+
+/**
+ * Renders an expandable wrapper for editing asset fields.
+ *
+ * @param {Object} props React props.
+ * @param {JSX.Element} props.heading Heading.
+ * @param {JSX.Element} props.subheading Subheading.
+ * @param {JSX.Element} props.help Help content to be shown after clicking on the help icon.
+ * @param {number} props.numOfIssues The number of issues used to label on UI. It only labels when the number is greater than 0.
+ * @param {boolean} [props.initialExpanded=false] Whether the UI to initial with expanded content.
+ * @param {boolean} [props.disabled=false] Whether display the UI in disabled style. It will collapse the content when disabled.
+ * @param {JSX.Element} [props.children] Content to be rendered.
+ * @param {import('react').MutableRefObject<AssetFieldHandler>} ref React ref to be attached to the handler of this component.
+ */
+function AssetField(
+	{
+		heading,
+		subheading,
+		help,
+		numOfIssues,
+		initialExpanded = false,
+		disabled = false,
+		children,
+	},
+	ref
+) {
+	const containerRef = useRef();
+	const [ isExpanded, setExpanded ] = useState( initialExpanded );
+
+	const isReducedMotion = useReducedMotion();
+
+	useImperativeHandle( ref, () => ( {
+		scrollIntoComponent() {
+			containerRef.current.scrollIntoView( {
+				behavior: isReducedMotion ? 'auto' : 'smooth',
+				inline: 'nearest',
+				block: 'nearest',
+			} );
+		},
+	} ) );
+
+	const handleToggle = () => {
+		setExpanded( ! isExpanded );
+	};
+
+	const issuePillText = sprintf(
+		// translators: %d: number of issues in a asset field.
+		_n( '%d issue', '%d issues', numOfIssues, 'google-listings-and-ads' ),
+		numOfIssues
+	);
+
+	const className = classnames(
+		'gla-asset-field',
+		disabled ? 'gla-asset-field--is-disabled' : false
+	);
+
+	const shouldExpanded = isExpanded && ! disabled;
```

Comment:
💅  Another renaming opinion.

Maybe `shouldExpand`, `shouldBeExpanded`, `isVisible` or `isExpanded` (this last one if we replace with the second option in my comment here https://github.com/woocommerce/google-listings-and-ads/pull/1853/files#r1072059197)

Reply:
Renamed in https://github.com/woocommerce/google-listings-and-ads/pull/1853/commits/4211d7cceb0d26d08d51c24c93a1124d1796e309.